hls4ml/writer/vivado_accelerator_writer.py

```diff
@@ -17,6 +17,10 @@ def write_axi_wrapper(self, model):
         inp_axi_t, out_axi_t, inp, out = self.vivado_accelerator_config.get_corrected_types()
         indent = '    '
 
+        io_type = model.config.get_config_value('IOType')
+        interface = model.config.get_config_value('AcceleratorConfig')['Interface'] if model.config.get_config_value('AcceleratorConfig') else None
+        config_weights = (io_type == 'io_stream') and (interface == 'axi_master')
```

Comment:
Same comment as before that I think a different trigger for reprogrammable weights is needed